docs/source/implicit_diff/implicit_diff.rst

```diff
@@ -10,15 +10,50 @@ Implicit Differentiation
     :width: 80%
     :align: center
 
-Implicit differentiation is the task of differentiating the solution of a minimization problem with respect to its inputs.
+Implicit differentiation is the task of differentiating through the solution of an optimization problem satisfying a mapping function :math:`T` capturing the optimality conditions of the problem.
+The simplest example is to differentiate through the solution of a minimization problem with respect to its inputs.
 Namely, given
 
 .. math::
 
     \boldsymbol{\theta}^{\prime} (\boldsymbol{\phi}) \triangleq \underset{\boldsymbol{\theta}}{\mathop{\operatorname{argmin}}} ~ \mathcal{L}^{\text{in}} (\boldsymbol{\phi},\boldsymbol{\theta}).
 
 By treating the solution :math:`\boldsymbol{\theta}^{\prime}` as an implicit function of :math:`\boldsymbol{\phi}`, the idea of implicit differentiation is to directly get analytical best-response derivatives :math:`\nabla_{\boldsymbol{\phi}} \boldsymbol{\theta}^{\prime} (\boldsymbol{\phi})` by the implicit function theorem.
-This is suitable for algorithms when the inner-level optimal solution is achieved :math:`\left. \frac{\partial \mathcal{L}^{\text{in}} (\boldsymbol{\phi}, \boldsymbol{\theta})}{\partial \boldsymbol{\theta}} \right\rvert_{\boldsymbol{\theta} = \boldsymbol{\theta}^{\prime}} = 0` (e.g., the function :math:`F` in the figure means the solution is obtained by unrolled gradient steps) or reaches some stationary conditions :math:`F (\boldsymbol{\phi}, \boldsymbol{\theta}^{\prime}) = 0`, such as `IMAML <https://arxiv.org/abs/1909.04630>`_ and `DEQ <https://arxiv.org/abs/1909.01377>`_.
+
```

Comment:
Can we do something like this. We use the equation T (\boldsymbol{\phi}, \boldsymbol{\theta}^{\prime}(\boldsymbol{\phi})) = \boldsymbol{0} and say if the solution of the inner-loop optimization is a root for such function, one can use implicit gradient.

Then we list two subsections, (need to split them up so it is clear) say for example, when we use gradient descent to solve for the optimization, the above equation is instantiated by the (first-order gradient =0). This is the subsection of IMAML.

Then the subsection of fixed-point, say DEQ, in this case the equation is instantiated by T-\theta = 0.

And also need to address that F is the iterative function, T is the stationary function at the beginning of these two subsections.